pkg/controller/worker/machines.go

```diff
@@ -163,7 +165,8 @@ func (w *workerDelegate) generateMachineConfig() error {
 				"secret": map[string]interface{}{
 					"cloudConfig": string(pool.UserData),
 				},
-				"blockDevices": blockDevices,
+				"blockDevices":            blockDevices,
+				"instanceMetadataOptions": instanceMetadataOptions,
```

Comment:
Nit: we can call `computeInstanceMetadata(workerConfig)` directly here.

Reply:
The only reason I didn't do it like this is because this line (`169`) is part of the per-zone loop. The metadata are common per worker group and not needed to be called per zone. 

It is a minuscule optimization that saves a function call but if you think that it improves readability we can follow your suggestion.

Reply:
Good point. No calculate the options for metadata service once per workerpool is sufficient.